index.js

```diff
@@ -3,47 +3,54 @@
 const { ServerResponse } = require('http')
 const fp = require('fastify-plugin')
 const WebSocket = require('ws')
-const findMyWay = require('find-my-way')
 
 const kWs = Symbol('ws')
 
 function fastifyWebsocket (fastify, opts, next) {
-  let handle = noHandle
-
-  if (opts.handle) {
-    if (typeof opts.handle !== 'function') {
-      return next(new Error('invalid handle function'))
+  let errorHandler = defaultErrorHandler
+  if (opts.errorHandler) {
+    if (typeof opts.errorHandler !== 'function') {
+      return next(new Error('invalid errorHandler function'))
     }
 
-    handle = wsHandle.bind(null, opts.handle)
+    errorHandler = opts.errorHandler
   }
 
   const options = Object.assign({}, opts.options)
   if (!options.server && !options.noServer) {
     options.server = fastify.server
   }
 
-  const router = findMyWay({
-    ignoreTrailingSlash: true,
-    defaultRoute: handle
-  })
-
   const wss = new WebSocket.Server(options)
   wss.on('connection', handleRouting)
 
   fastify.decorate('websocketServer', wss)
 
+  fastify.addHook('onError', (request, reply, error, done) => {
+    if (request.raw[kWs]) {
+      // Hijack reply to prevent fastify from sending the error after onError hooks are done running
+      reply.hijack()
+      // Handle the error
+      errorHandler.call(this, error, request.raw[kWs], request, reply)
+    }
+    done()
```

Comment:
I think this should be called with error as an argument in case it's not a websocket.

Reply:
Do you mean passing the error to the `done` function?

Quote [from](https://www.fastify.io/docs/master/Hooks/#onerror):

> Notice: unlike the other hooks, pass an error to the done function is not supported.

Reply:
You are right, forget about it!